File: exam-slammr-webapp/src/main/java/uk/co/claritysoftware/exam/slammr/webapp/web/controller/LoginPageController.java
package uk.co.claritysoftware.exam.slammr.webapp.web.controller;

import java.security.Principal;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.servlet.ModelAndView;
import uk.co.claritysoftware.exam.slammr.webapp.web.config.SocialProviders;

/**
 * Controller to handle login page requests
 */
@Controller
@RequestMapping("/login")
public class LoginPageController {

	private final SocialProviders socialProviders;

	@Autowired
	public LoginPageController(SocialProviders socialProviders) {
		this.socialProviders = socialProviders;
	}

	@GetMapping
	public ModelAndView getLoginPage(Principal principal) {
		if (principal != null) {
			return new ModelAndView("redirect:/");
		} else {
			return new ModelAndView("auth/login")
					.addObject("facebookConfigured", socialProviders.isFacebookConfigured())
					.addObject("twitterConfigured", socialProviders.isTwitterConfigured())
					.addObject("linkedInConfigured", socialProviders.isLinkedinConfigured());
		}
	}
}

File: exam-slammr-webapp/src/main/kotlin/uk/co/claritysoftware/exam/slammr/webapp/web/controller/LoginPageController.kt
package uk.co.claritysoftware.exam.slammr.webapp.web.controller

import org.slf4j.LoggerFactory
import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.servlet.ModelAndView
import uk.co.claritysoftware.exam.slammr.webapp.web.config.SocialProviders
import java.security.Principal

/**
 * Controller to handle login page requests
 */
@Controller
@RequestMapping("/login")
class LoginPageController(private val socialProviders: SocialProviders) {

	private companion object {
		val log = LoggerFactory.getLogger(LoginPageController::class.java)
	}

	@GetMapping
	fun getLoginPage(principal: Principal?): ModelAndView {

		return principal?.let {
			ModelAndView("redirect:/")
		}
				?: ModelAndView("auth/login")
						.addObject("facebookConfigured", socialProviders.isFacebookConfigured)
						.addObject("twitterConfigured", socialProviders.isTwitterConfigured)
						.addObject("linkedInConfigured", socialProviders.isLinkedinConfigured)
	}

}